packages/libs/client/src/signing/eckoWallet/eckoCommon.ts

```diff
@@ -0,0 +1,46 @@
+import { IEckoConnectOrStatusResponse } from './eckoTypes';
+
+export const isInstalled = (): boolean => {
+  const { kadena } = window;
+  return Boolean(kadena && kadena.isKadena);
+};
+
+export const isConnected = async (networkId: string): Promise<boolean> => {
+  if (!isInstalled()) {
+    return false;
+  }
+
+  const checkStatusResponse =
+    await window.kadena?.request<IEckoConnectOrStatusResponse>({
+      method: 'kda_checkStatus',
+      networkId,
+    });
+
+  if (checkStatusResponse?.status === 'fail') {
+    return false;
+  }
+
+  return true;
```

Comment:
Is this the best way to check for active status or should we return true on a positive signal?

Reply:
Although Ecko returns a 'fail' status when it isn't connected, I agree that it might be better to turn things around to only return `true` when the returned status is `success`. Then it'll also return false when Ecko returns `success`.